installer/Helpers/RequirementsChecker.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Installer\Helpers;
+
+class RequirementsChecker
+{
+	/**
+	 * Minimum PHP Version Supported (Override is in installer.php config file).
+	 *
+	 * @var _minPhpVersion
+	 */
+	private $_minPhpVersion = '7.2.0';
+
+	/**
+	 * Check for the server requirements.
+	 *
+	 * @param array $requirements
+	 *
+	 * @return array
+	 */
+	public function check(array $requirements)
+	{
+		$results = [];
+		foreach ($requirements as $type => $requirement_) {
+			switch ($type) {
+				// check php requirements
+				case 'php':
+					foreach ($requirements[$type] as $requirement) {
+						$results['requirements'][$type][$requirement] = true;
+						if (!extension_loaded($requirement)) {
+							$results['requirements'][$type][$requirement]
+								= false;
+							$results['errors'] = true;
+						}
+					}
+
+					if ($this->checkExec()) {
+						$results['requirements'][$type]['Php exec() available'] = true;
```

Comment:
Is it worth tracking the negative here?

Reply:
I would say yes. Because it tells the user why he cannot install it: we need the exec to apply the migration. ;)

Reply:
I think I meant displaying the result if `exec` is enabled. In hindsight I think not.

Reply:
Ah, I see what you mean.
Well the way I see it is that we display the actual dependencies, and show that they are satisfied.
I would rather see what dependencies I need and are checked. Rather than just a message "All dependencies are satisfied, you are good to go, Sir." :smiley: